src/PointwiseFunctions/Elasticity/PotentialEnergy.cpp

```diff
@@ -0,0 +1,73 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/Elasticity/PotentialEnergy.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Tags.hpp"
+#include "Elliptic/Systems/Elasticity/Tags.hpp"
+#include "PointwiseFunctions/Elasticity/ConstitutiveRelations/ConstitutiveRelation.hpp"
+#include "PointwiseFunctions/Elasticity/ConstitutiveRelations/Tags.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace Elasticity {
+
+template <size_t Dim>
+void potential_energy_density(
+    const gsl::not_null<Scalar<DataVector>*> potential_energy_density,
+    const tnsr::ii<DataVector, Dim>& strain,
+    const tnsr::I<DataVector, Dim>& coordinates,
+    const ConstitutiveRelations::ConstitutiveRelation<Dim>&
+        constitutive_relation) noexcept {
+  *potential_energy_density =
+      make_with_value<Scalar<DataVector>>(coordinates, 0.);
```

Comment:
`destructive_resize_components(potential_energy_density, coordinates.begin()->size())` and then initialize to 0.  (The point of taking by reference is to avoid the allocation when possible.)

Reply:
And then remove this statement.

Reply:
Ah sorry about that! Somewhere between running the test and checking the documentation I must have reverted the deletion.